File: Task/Sequence-of-non-squares/C-sharp/sequence-of-non-squares.cs
using System;
using System.Diagnostics;

namespace sons
{
    class Program
    {
        static void Main(string[] args)
        {
            for (int i = 1; i < 23; i++)
                Console.WriteLine(nonsqr(i));

            for (int i = 1; i < 1000000; i++)
            {
                double j = Math.Sqrt(nonsqr(i));
                Debug.Assert(j != Math.Floor(j),"Square");
            }
        }

        static int nonsqr(int i)
        {
            return (int)(i + Math.Floor(0.5 + Math.Sqrt(i)));
        }
    }
}

File: Task/Sequence-of-non-squares/Delphi/sequence-of-non-squares.pas
program Sequence_of_non_squares;

uses
  System.SysUtils, System.Math;

function nonsqr(i: Integer): Integer;
begin
  Result := Trunc(i + Floor(0.5 + Sqrt(i)));
end;

var
  i: Integer;
  j: Double;

begin

  for i := 1 to 22 do
    write(nonsqr(i), ' ');
  Writeln;

  for i := 1 to 999999 do
  begin
    j := Sqrt(nonsqr(i));
    if (j = Floor(j)) then
      Writeln(i, 'Is Square');
  end;
end.
